test/functional/launch/reduce-basic/tests/test-launch-basic-ReduceBitAnd.hpp

```diff
@@ -0,0 +1,198 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_BASIC_REDUCEBITAND_HPP__
+#define __TEST_LAUNCH_BASIC_REDUCEBITAND_HPP__
+
+#include <cstdlib>
+#include <ctime>
+#include <numeric>
+#include <vector>
+
+template <typename IDX_TYPE, typename DATA_TYPE,
+          typename SEG_TYPE,
+          typename LAUNCH_POLICY,
+          typename GLOBAL_THREAD_POLICY,
+          typename REDUCE_POLICY>
+void LaunchReduceBitAndBasicTestImpl(const SEG_TYPE& seg,
+                                     const std::vector<IDX_TYPE>& seg_idx,
+                                     camp::resources::Resource working_res)
+{
+  IDX_TYPE data_len = seg_idx[seg_idx.size() - 1] + 1;
+  IDX_TYPE idx_len = static_cast<IDX_TYPE>( seg_idx.size() );
+
+  DATA_TYPE* working_array;
+  DATA_TYPE* check_array;
+  DATA_TYPE* test_array;
+
+  constexpr int threads = 256;
+  int blocks = (seg.size() - 1)/threads + 1;
+
+  allocateForallTestData<DATA_TYPE>(data_len,
+                                    working_res,
+                                    &working_array,
+                                    &check_array,
+                                    &test_array);
+
+  //
+  // First a simple non-trivial test that is mildly interesting
+  //
+  for (IDX_TYPE i = 0; i < data_len; ++i) {
+    test_array[i] = 13;
+  }
+  working_res.memcpy(working_array, test_array, sizeof(DATA_TYPE) * data_len);
+
+  RAJA::ReduceBitAnd<REDUCE_POLICY, DATA_TYPE> simpand(21);
+
+  RAJA::launch<LAUNCH_POLICY>
+    (RAJA::LaunchParams(RAJA::Teams(blocks), RAJA::Threads(threads)),
+     [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
```

Comment:
Just a question: I've been using `[&]` to capture the local variables by reference, per the original design of `RAJA::Teams`. Did something change where now the best practice for Launch is to capture by value, or is capture by value just used for these simple tests?

Reply:
I believe capture-by-value is used on the outer lambda passed to the RAJA::launch method, and lambdas internal to that use capture-by-reference.  @artv3 please correct me if I'm wrong.

Reply:
The outer lambda is capture by value, inner lambdas are capture by reference.

Reply:
Thanks, that makes sense.